libsolidity/analysis/TypeChecker.cpp

```diff
@@ -694,6 +694,7 @@ bool TypeChecker::visit(VariableDeclaration const& _variable)
 	if (
 		m_scope->contractKind() == ContractDefinition::ContractKind::Interface
 		&& !_variable.isCallableParameter()
+		&& !m_insideStruct
```

Comment:
The comment above this needs to be adjusted.

Reply:
Not related to this PR, but the comment below is outdated as well and still talks about ``var`` and the old scoping rules...

Reply:
didn't read the comment when touching the if-expr. I've adapted that now.

Reply:
We should keep the comment below in mind as well - not sure whether it's worth a whole issue, though :-). Can you create a small PR for it or shall I do that or maybe we can just adjust it right away in this PR while we're at it?